File: app/src/main/java/com/zjw/appmethodtime/MainActivity.java
package com.zjw.appmethodtime;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.zjw.mylibrary.LibActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    protected TextView mTextView;
    private String TAG = "TAG";

    @CostTime
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        super.setContentView(R.layout.activity_main);
        initView();
    }

    @CostTime
    private void initView() {
        mTextView = (TextView) findViewById(R.id.text_view);
        mTextView.setOnClickListener(MainActivity.this);
    }

    @CostTime
    @Override
    public void onClick(View view) {
        if (view.getId() == R.id.text_view) {
            Log.e(TAG, "onClick");
            final Intent intent = new Intent(MainActivity.this, LibActivity.class);
            startActivity(intent);
        }
    }
}

File: app/src/main/java/com/zjw/appmethodtime/MainActivity.kt
package com.zjw.appmethodtime

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private val TAG = "TAG"

    @CostTime
    override fun onCreate(savedInstanceState: Bundle?) {
        val startTime = System.nanoTime()
        super.onCreate(savedInstanceState)
        super.setContentView(R.layout.activity_main)
        initView()
        val endTime = System.nanoTime()
        Log.i("TAG","onCreate cost is "+(endTime - startTime)/1000000f)
    }

    @CostTime
    private fun initView() {
        mTextView.setOnClickListener(this@MainActivity)
        //android官方 lint插件自带的warring提示
        //自定义MyLogDetector lint error提示
        // Log.e(TAG,"test")

    }

    @CostTime
    override fun onClick(view: View) {
        if (view.id == R.id.mTextView) {
            try {
                Thread.sleep(500)
            } catch (e: Exception) {
            }
        }
    }
}
